spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/table/SparkBaseTable.java

```diff
@@ -91,12 +90,10 @@ public StructType schema() {
 
   @Override
   public Map<String, String> properties() {
-    Map properties = new HashMap();
-    if (gravitinoTable.properties() != null) {
-      properties.putAll(gravitinoTable.properties());
-    }
-
-    properties = propertiesConverter.toSparkTableProperties(properties);
+    Map<String, String> properties;
+    properties =
+        propertiesConverter.toSparkTableProperties(
+            gravitinoTable.properties(), getSparkTable().properties());
```

Comment:
This seems hacky to combine properties from Gravitino and realCatalog. I think the right direction is to provide the properties from Gravitino.

Reply:
do you know the reason why couldn't get the reserved properties from Gravitino?

Reply:
> This seems hacky to combine properties from Gravitino and realCatalog. I think the right direction is try to provide the properties from Gravitino.

My original implementation was this, but I was concerned that if all reserved properties were placed into `Gravitino` `IcebergTable`'s properties, there would be differences between different computing engines, such as `Spark` and `Flink`.
Because they all will get the same reserved properties from the `Gravitino` `IcebergTable`.
For this reason, i changed the solution to retrieve reserved properties from the realTable at the spark-connector. cc @FANNG1

Reply:
> > This seems hacky to combine properties from Gravitino and realCatalog. I think the right direction is try to provide the properties from Gravitino.
> 
> My original implementation was this, but I was concerned that if all reserved properties were placed into `Gravitino` `IcebergTable`'s properties, there would be differences between different computing engines, such as `Spark` and `Flink`. Because they all will get the same reserved properties from the `Gravitino` `IcebergTable`. For this reason, i changed the solution to retrieve reserved properties from the realTable at the spark-connector. cc @FANNG1

@FANNG1 Should i add the reserved properties into the Gravitino IceebrgTable's properties?

Reply:
> do you know the reason why couldn't get the reserved properties from Gravitino?

@caican00 do you know the reason?

Reply:
> > do you know the reason why couldn't get the reserved properties from Gravitino?
> 
> @caican00 do you know the reason?

@FANNG1 I think that we can get  reserved properties from Gravitino finally.
The reason why i don't get reserved properties from Gravitino right now is that some capabilities are not yet implemented on the Gravitino side, such as `Snapshot`. So i cannot get the `current-shanpshot-id `directly from Gravitino.
And it may take more time to implement `Snapshot` on Gravitino side.
To get the `location` quickly, I chose to implement this at the spark-connector side.

Another idea is that instead of getting all reserved properties at once, we can get location first to support iceberg-partition PR.

we can refactor the `toSparkTableProperties(Map<String, String> properties)`, change it to `Map<String, String> toSparkTableProperties(com.datastrato.gravitino.rel. Table gravitinoTable)`,  and then we can get the `location` from gravitinoTable

Reply:
WDYT?

Reply:
I prefer to delay the support of reserved properties,  For `location` how could we retrieve it from GravitinoTable?

Reply:
seems we could add `location` to `properteis` in https://github.com/datastrato/gravitino/blob/main/catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergTable.java#L133

Reply:
> seems we could add `location` to `properteis` in https://github.com/datastrato/gravitino/blob/main/catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergTable.java#L133

@FANNG1 
This involves the support of other reserved properties later. should we also put other reserved properties into the Table's properties on Gravitino side?
in Iceberg, `location` is not stored in `Table`'s properties in the real `Iceberg Table` , but is retrieved from the variable of the real `Iceberg Table` at the connector side.

Reply:
> I prefer to delay the support of reserved properties, For `location` how could we retrieve it from GravitinoTable?

i would like to add `IcebergTable` of Gravitino into the `bundled-catalog`, and then we can get the `IcebergTable` at spark-connector side. and then we can add a  `@Getter` for `location` of `IcebergTable`, after these, we can get the `location` from gravitinoTable finally.

Reply:
But now I don't know how to add `IcebergTable` to `bundled-catalog`, do you know this?

Reply:
IMO we could delay the support of other properties,  For `location`, we could add it in Gravitino server side in partition PR, WDYT?

Reply:
`bundled-catalog` is designed to store properties meta not IcebergTable

Reply:
> IMO we could delay the support of other properties, For `location`, we could add it in Gravitino server side in partition PR, WDYT?

ok for me.

Reply:
> `bundled-catalog` is designed to store properties meta not IcebergTable

got it.  Any ideas about supporting other reserved properties?  Just in time to discuss.
it seems that if the `IcebergTable` is not available on the spark-conenctor side, we must put all reserved properties into `IcebergTable`'s properties. cc @FANNG1

Reply:
any other properties, we could rebuild properties like Iceberg spark table in Gravitino server https://github.com/apache/iceberg/blob/main/spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java#L213-L249

Reply:
@FANNG1 putting location into properties in Gravitino side will cause unexpected problems.
https://github.com/datastrato/gravitino/actions/runs/8718294890/job/23915239099?pr=2709

Reply:
![image](https://github.com/datastrato/gravitino/assets/94670132/d6560167-3a51-4364-8c49-26de38ada983)

Reply:
@yuqi1129 @diqiu50 could help to point out how to fix trino it after adding location properties for Iceberg table.

Reply:
@caican00 
You can directly change the file `lakehouse-iceberg/00000_create_table.txt` and use the correct output. 

Use the wildcard character '%' if necessary. 
<img width="1059" alt="image" src="https://github.com/datastrato/gravitino/assets/15794564/9a942709-4e45-47b0-b719-f18f2b4a69bc">

Reply:
> @caican00 You can directly change the file `lakehouse-iceberg/00000_create_table.txt` and use the correct output.
> 
> Use the wildcard character '%' if necessary. <img alt="image" width="1059" src="https://private-user-images.githubusercontent.com/15794564/323151214-9a942709-4e45-47b0-b719-f18f2b4a69bc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMzNDQ1MTUsIm5iZiI6MTcxMzM0NDIxNSwicGF0aCI6Ii8xNTc5NDU2NC8zMjMxNTEyMTQtOWE5NDI3MDktNGU0NS00N2IwLWI3MTktZjE4ZjJiNGE2OWJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE3VDA4NTY1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjYWQxMzI1OTlkNWJmNDhlYWE2OTIzNzJkNjM3ZDQwZWY2MjQ0ZDU2YmMzM2FmMTUwNGI4OGUxMTkyOGFjZmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Xax_9GLGaOdcYqLLu5qLsm16hW35j-bhooXiQHQXVKA">

@yuqi1129 It seems that executing trino's showCreateTable command without Gravitino does not show location in its properties, which seems inconsistent with the original behavior.  Is that reasonable?

Reply:
> @caican00 You can directly change the file `lakehouse-iceberg/00000_create_table.txt` and use the correct output.
> 
> Use the wildcard character '%' if necessary. <img alt="image" width="1059" src="https://private-user-images.githubusercontent.com/15794564/323151214-9a942709-4e45-47b0-b719-f18f2b4a69bc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMzNDU2NzUsIm5iZiI6MTcxMzM0NTM3NSwicGF0aCI6Ii8xNTc5NDU2NC8zMjMxNTEyMTQtOWE5NDI3MDktNGU0NS00N2IwLWI3MTktZjE4ZjJiNGE2OWJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE3VDA5MTYxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEyYzNhMjJiMGFjM2MwNTM1MTczNTI0OWYyN2JjYzgxYzVkMGJhMmZlNTVmMTU1MTAzOWZlNGY5ZThjNzBkZDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qRb4peDf8uO48TPdw2teik01vkQ1SgbAyE05iV1_P9o">

got it.